### PR TITLE
fix sign function and add corresponding unittest

### PR DESCRIPTION
**Pull request by hsinhoyeh**
this PR is for #6, thanks

### CODE REVIEW

File: generator_test.go [hunk 1]
```diff
@@ -116,3 +116,27 @@ func TestGenerateClaim(t *testing.T) {
 		t.Fatalf("Unexpected claim\nExpect:\t%s\nGot:\t%s", expect, got)
 	}
 }
+
+func TestSign(t *testing.T) {
+	for _, g := range []struct {
```
Comment (MrGossett) [inline generator_test.go]: If there's just one case to test, then let's drop the loop.

File: generator_test.go [hunk 2]
```diff
@@ -116,3 +116,27 @@ func TestGenerateClaim(t *testing.T) {
...
+		key string
+		in  string
+		out string
+	}{
+		{
+			key: "you cannot see me",
+			in:  "the winter is comming",
```
Comment (zabawaba99) [inline generator_test.go]: I see what you did there. :+1:

File: generator_test.go (hunk 1 above)
Comment (zabawaba99) [inline generator_test.go]: Agreed.

Comment (hsinhoyeh) [inline generator_test.go]: done

File: generator_test.go (hunk 2 above)
Comment (hsinhoyeh) [inline generator_test.go]: (Y)

**Comment by zabawaba99** [comment]
Looks fine to me.

**Comment by MrGossett** [comment]
:+1:

**Pull request merged**